controllers/imagejob/imagejob_controller.go

```diff
@@ -365,11 +398,34 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1.ImageJ
 			return err
 		}
 		log.Info("Started "+containerName+" pod on node", "nodeName", nodeName)
+
+		// send pod to podQueue to create thread and wait for PodRunning phase
+		waitGroup.Add(1)
+		podQueue <- pod
 	}
 
+	// wait for all PodRunning threads to complete
+	waitGroup.Wait()
+
 	return nil
 }
 
+func (r *Reconciler) isPodRunning(podName, podNamespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		currentPod := &corev1.Pod{}
+		namespacedName := types.NamespacedName{
+			Namespace: podNamespace,
+			Name:      podName,
+		}
+
+		if err := r.Get(context.TODO(), namespacedName, currentPod); err != nil {
+			return false, client.IgnoreNotFound(err)
+		}
+
+		return currentPod.Status.Phase == corev1.PodPhase(corev1.PodRunning) || currentPod.Status.Phase == corev1.PodPhase(corev1.PodSucceeded), nil
```

Comment:
What happens here if the phase is PodFailed?

Reply:
thats a good point, I can change this method to just check that the status is no longer in pending?